staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -230,6 +230,7 @@ type Cacher struct {
 	// See: https://golang.org/pkg/sync/atomic/ for more information
 	incomingHWM storage.HighWaterMark
```

Comment:
We might want to document this further as indicated in the issue:

We `HighWaterMark` the length of the incoming `watchEvent`s channel. Our intent is mainly to keep track of the highest value of this length so that in case we need to debug cases where the channel is getting backed up and events from it are not being popped and removed, we have logs for occurrences of when the highest channel lengths were achieved.

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L799

Reply:
Can we add this line to the documentation:
`The word "incoming" refers to the incoming 'watchEvent's channel length. We HighWaterMark it keep track of the highest value of this length so that in case we need to debug cases where the channel is getting backed up and events from it are not being popped and removed, we have logs for occurrences of when the highest channel lengths were achieved.`

Reply:
I feel since the issue needed a complete context of incomingHWM , maybe condensing it further will not be beneficial.